runtime/lua/vim/lsp/util.lua

```diff
@@ -1795,8 +1795,18 @@ function M.locations_to_items(locations, offset_encoding)
       local row = pos.line
       local end_row = end_pos.line
       local line = lines[row] or ''
-      local col = M._str_byteindex_enc(line, pos.character, offset_encoding)
-      local end_col = M._str_byteindex_enc(lines[end_row] or '', end_pos.character, offset_encoding)
+      local end_line = lines[end_row] or ''
+      ---@type boolean, integer, integer
+      local ok, col, end_col
+      ok, col = pcall(_str_byteindex_enc, line, pos.character, offset_encoding)
```

Comment:
instead of swallowing errors with pcall() is there an easy way to check that the length is valid? then the code would be clearer and more direct:

```lua
local col = pos.character <= ... and M._str_byteindex_enc(line, pos.character, offset_encoding) or (vim.fn.strcharlen(line) - 1)
```

Reply:
I can check if pos.character is larger than the `vim.fn.strcharlen()` of the string but i was trying to avoid calling the len function until it was needed

Reply:
true. i think it's probably worth it though. pcall() can hide other kinds of errors, which would make other bugs hard to find.